src/filemenu/filemenu_pal.c

```diff
@@ -0,0 +1,377 @@
+#include "common.h"
+#include "filemenu.h"
+#include "message_ids.h"
+#include "hud_element.h"
+#include "ld_addrs.h"
+
+extern HudScript HES_Spirit1;
+
+HudScript* D_filemenu_8024F120[] = {&HES_Spirit1};
+
+u32 D_filemenu_8024F124 = 0x00000000;
+
+s32 D_filemenu_8024F128 = 0x00000000;
+
+u8 D_filemenu_8024F12C[] = {0x00, 0x01, 0x02, 0x03};
+
+u32 D_filemenu_8024F130[] = {
+    MSG_PAL_Menu_0056,
+    MSG_PAL_Menu_0057,
+    MSG_PAL_Menu_0058,
+    MSG_PAL_Menu_0059,
+};
+
+Gfx D_filemenu_8024F140[] = {
+    gsDPPipeSync(),
+    gsDPSetCycleType(G_CYC_1CYCLE),
+    gsDPSetRenderMode(G_RM_CLD_SURF, G_RM_CLD_SURF2),
+    gsDPSetTextureFilter(G_TF_POINT),
+    gsSPTexture(0xFFFF, 0xFFFF, 0, G_TX_RENDERTILE, G_ON),
+    gsDPSetTexturePersp(G_TP_NONE),
+    gsDPSetColorDither(G_CD_DISABLE),
+    gsDPSetAlphaDither(G_AD_DISABLE),
+    gsDPSetTextureLOD(G_TL_TILE),
+    gsDPSetTextureLUT(G_TT_NONE),
+    gsDPSetTextureDetail(G_TD_CLAMP),
+    gsDPSetTextureConvert(G_TC_FILT),
+    gsDPSetCombineKey(G_CK_NONE),
+    gsDPSetAlphaCompare(G_AC_NONE),
+    gsDPNoOp(),
+    gsDPSetScissor(G_SC_NON_INTERLACE, 0, 0, 320, 240),
+    gsSPEndDisplayList(),
+};
+
+u32 D_filemenu_8024F1C8[] = {
+    0x00000300,
+    0x00000000,
+    0x00000200,
+    0x00000100,
+};
+
+void filemenu_draw_pal_8024d6a0(MenuPanel* menu, s32 baseX, s32 baseY, s32 width, s32 height, s32 opacity, s32 darkening);
+void filemenu_draw_pal_8024DAA0(MenuPanel* menu, s32 baseX, s32 baseY, s32 width, s32 height, s32 opacity, s32 darkening);
+void filemenu_update_pal_80247f40(s32 windowIndex, s32* flags, s32* posX, s32* posY, s32* posZ, f32* scaleX, f32* scaleY,
+                                  f32* rotX, f32* rotY, f32* rotZ, s32* darkening, s32* opacity);
+void filemenu_draw_pal_8024DA50(MenuPanel* menu, s32 baseX, s32 baseY, s32 width, s32 height, s32 opacity, s32 darkening);
+
+void filemenu_draw_pal_8024DA00(MenuPanel* menu, s32 baseX, s32 baseY, s32 width, s32 height, s32 opacity, s32 darkening);
+void filemenu_draw_pal_8024D9B0(MenuPanel* menu, s32 baseX, s32 baseY, s32 width, s32 height, s32 opacity, s32 darkening);
+void filemenu_pal_handle_input(MenuPanel *);
+void filemenu_pal_init(MenuPanel *);
+void filemenu_pal_cleanup(MenuPanel *);
+void filemenu_pal_update(MenuPanel *);
+
+MenuWindowBP D_filemenu_8024F1D8[] = {
+    {
+        .windowID = WINDOW_ID_FILEMENU_TITLE,
+        .unk_01 = 0,
+        .pos = { .x = 0, .y = 0 },
+        .width = 0,
+        .height = 0,
+        .priority = WINDOW_PRIORITY_0,
+        .fpDrawContents = &filemenu_draw_pal_8024d6a0,
+        .tab = NULL,
+        .parentID = WINDOW_ID_FILEMENU_MAIN,
+        .fpUpdate = { .func=&filemenu_update_show_title },
+        .extraFlags = 0,
+        .style = { .customStyle = &filemenu_windowStyles[3] }
+    },
+
```

Comment:
Why is there an empty line between some of these

Reply:
Leftover from it being not an array